### ISSUE TITLE
Stacked bar labels incorrect due to alphabetization

### ISSUE THREAD

**Issue by kbeck527** [ropensci/plotly#662]
When plotting a stacked bar plot, the hover labels are being incorrectly displayed. It appears to me that ggplot creates a cumulative display value for the stacked bar plot as noted in issue #557 (this behavior may not be what others want, but it's at least accurate). If the labels for the stacked bars are not in order (see example below), there is a second issue with the hover text. When applying ggplotly to the plot object, plotly reorders the bars by alphabetizing but maintains the earlier cumulative value label from ggplot. This makes the labels entirely useless. Here is a reproducible example:

```
library(reshape2)
library(ggplot2)
library(plotly)


df = data.frame("name" = c("A", "B", "C", "D", "A1"),
                "rep1" = c(37, 19, 93, 13, 121),
                "rep2" = c(26, 1, 54, 7, 123),
                "rep3" = c(21, 10, 34, 14, 211)
                )

melted_df = melt(df, variable.name = "sample", value.name = "counts")

p = ggplot(melted_df, aes(x = sample)) +
  geom_bar(aes(y = counts, fill = name), stat = "identity")
ggplotly(p)
```

Notice that the A1 values are now at the bottom of the plot but exhibit a value of 283 for example for rep1. However, the location of the bar is below the 200 tick mark on the y-axis.

Any help with this would be greatly appreciated.

```
> devtools::session_info()
Session info ----------------------------------------------------------------------------
 setting  value                       
 version  R version 3.2.4 (2016-03-10)
 system   x86_64, darwin13.4.0        
 ui       RStudio (0.99.896)          
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       America/Los_Angeles         
 date     2016-07-29                  

Packages --------------------------------------------------------------------------------
 package      * version    date       source                            
 assertthat     0.1        2013-12-06 CRAN (R 3.2.0)                    
 base64enc      0.1-3      2015-07-28 CRAN (R 3.2.0)                    
 beakR        * 0.0.0.9000 2016-07-22 local (metagenomics/mcaw-beakr@NA)
 class          7.3-14     2015-08-30 CRAN (R 3.2.4)                    
 cluster        2.0.4      2016-04-18 CRAN (R 3.2.5)                    
 colorspace     1.2-6      2015-03-11 CRAN (R 3.2.0)                    
 d3heatmap    * 0.6.1.1    2016-02-23 CRAN (R 3.2.3)                    
 dendextend     1.2.0      2016-06-22 CRAN (R 3.2.5)                    
 DEoptimR       1.0-6      2016-07-06 CRAN (R 3.2.5)                    
 devtools       1.12.0     2016-06-24 CRAN (R 3.2.5)                    
 digest         0.6.9      2016-01-08 CRAN (R 3.2.3)                    
 diptest        0.75-7     2015-06-08 CRAN (R 3.2.0)                    
 evaluate       0.9        2016-04-29 CRAN (R 3.2.5)                    
 flexmix        2.3-13     2015-01-17 CRAN (R 3.2.0)                    
 fpc            2.1-10     2015-08-14 CRAN (R 3.2.0)                    
 ggplot2      * 2.1.0      2016-03-01 CRAN (R 3.2.4)                    
 gridExtra      2.2.1      2016-02-29 CRAN (R 3.2.4)                    
 gtable         0.2.0      2016-02-26 CRAN (R 3.2.3)                    
 heatmaply    * 0.3.2      2016-05-26 CRAN (R 3.2.5)                    
 htmltools    * 0.3.5      2016-03-21 CRAN (R 3.2.4)                    
 htmlwidgets    0.6        2016-02-25 CRAN (R 3.2.3)                    
 httr           1.2.1      2016-07-03 CRAN (R 3.2.5)                    
 jsonlite       1.0        2016-07-01 CRAN (R 3.2.5)                    
 kernlab        0.9-24     2016-03-29 CRAN (R 3.2.4)                    
 knitr          1.13       2016-05-09 CRAN (R 3.2.5)                    
 labeling       0.3        2014-08-23 CRAN (R 3.2.0)                    
 lattice        0.20-33    2015-07-14 CRAN (R 3.2.4)                    
 magrittr       1.5        2014-11-22 CRAN (R 3.2.0)                    
 MASS           7.3-45     2015-11-10 CRAN (R 3.2.4)                    
 mclust         5.2        2016-03-31 CRAN (R 3.2.4)                    
 memoise        1.0.0      2016-01-29 CRAN (R 3.2.3)                    
 modeltools     0.2-21     2013-09-02 CRAN (R 3.2.0)                    
 munsell        0.4.3      2016-02-13 CRAN (R 3.2.3)                    
 mvtnorm        1.0-5      2016-02-02 CRAN (R 3.2.3)                    
 nnet           7.3-12     2016-02-02 CRAN (R 3.2.4)                    
 plotly       * 3.6.0      2016-05-18 CRAN (R 3.2.5)                    
 plyr           1.8.4      2016-06-08 CRAN (R 3.2.5)                    
 png            0.1-7      2013-12-03 CRAN (R 3.2.0)                    
 prabclus       2.2-6      2015-01-14 CRAN (R 3.2.0)                    
 R6             2.1.2      2016-01-26 CRAN (R 3.2.3)                    
 RColorBrewer   1.1-2      2014-12-07 CRAN (R 3.2.0)                    
 Rcpp           0.12.5     2016-05-14 CRAN (R 3.2.5)                    
 reshape2     * 1.4.1      2014-12-06 CRAN (R 3.2.0)                    
 rmarkdown      1.0        2016-07-08 CRAN (R 3.2.5)                    
 robustbase     0.92-6     2016-05-31 CRAN (R 3.2.5)                    
 rstudioapi     0.6        2016-06-27 CRAN (R 3.2.5)                    
 scales       * 0.4.0      2016-02-26 CRAN (R 3.2.3)                    
 stringi        1.1.1      2016-05-27 CRAN (R 3.2.5)                    
 stringr      * 1.0.0      2015-04-30 CRAN (R 3.2.0)                    
 tibble         1.1        2016-07-04 CRAN (R 3.2.5)                    
 tidyr          0.5.1      2016-06-14 CRAN (R 3.2.5)                    
 trimcluster    0.1-2      2012-10-29 CRAN (R 3.2.0)                    
 viridis      * 0.3.4      2016-03-12 CRAN (R 3.2.4)                    
 whisker        0.3-2      2013-04-28 CRAN (R 3.2.0)                    
 withr          1.0.2      2016-06-20 CRAN (R 3.2.5)                    
 yaml           2.1.13     2014-06-12 CRAN (R 3.2.0) 
```

**Comment by cpsievert** [comment]
yikes, thanks for the detailed report!

**Comment by kbeck527** [comment]
You're welcome. Please let me know if there is any other info I can provide to help with the resolution of this bug.

**Comment by CollierKing** [comment]
I am having the same issue.  Any timeline for a fix? Loving everything that Plotly can do so far.  Thanks!

**Comment by cpsievert** [comment]
Fixed via 27bbdc1
